### DISCUSSION TITLE
Github action is present in every PR instead of the PR to the branch set in yml

### DISCUSSION THREAD

**Discussion by ostsmolynets** [community/community#26955] [Actions]
<p>I have github actions setup in yml like so:</p>
<p><img src="https://user-images.githubusercontent.com/62648334/181100945-789ccf14-1eba-42e9-8140-7322dc771c28.png" alt="develop_PR_jobs.png" data-base62-sha1="nuHjkA3mu6EHlWRRKCmnEIGIKAI" width="476" height="300"></p>
<p>When I created three PRs from the same branch (which was branched out of master) to master, rc, and develop the action above is present in all the PR instead of only being present in develop PR. Did I set up something wrong?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/ostsmolynets">@ostsmolynets</a> , </p>
<p>I set up similar scenario in my repo. The workflow run trigged by on pull_request event from source branch to develop branch are attached to all the PRs checks list. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/30131dae7e6617c8f2636897f9a8356113dbcff6.png" data-download-href="/uploads/short-url/6RhScOInKiBPtnRQdHuQs9CvMkm.png?dl=1" title="PR checks.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181100947-b2ce768b-a05d-4e12-bd95-52bedebd60d2.png" alt="PR checks.png" data-base62-sha1="6RhScOInKiBPtnRQdHuQs9CvMkm" width="690" height="480" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/30131dae7e6617c8f2636897f9a8356113dbcff6_2_690x480.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/30131dae7e6617c8f2636897f9a8356113dbcff6_2_1035x720.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/3/30131dae7e6617c8f2636897f9a8356113dbcff6.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/30131dae7e6617c8f2636897f9a8356113dbcff6_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">PR checks.png</span><span class="informations">1141×794 72.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p><strong>The check runs for the last commit will show in the Pull Request checks list.</strong> As the last commit is on source branch and all your PRs are from source branch to others, so the commit is associated to all the PRs and the workflow run is present in all the PRs. This is an expected behavior. </p>

**Reply by jaggaer-mdurcik** [reply]
Why is this expected?
We have checks in our workflows that validates  that combination of source and destination branch names in PR are allowed.
For example  a fix/XXXX branch might be allowed to be merged into dev but not into master. 
So when I create several PRs form the same branch/commit, I expect that the checks are executed separately for each PR and deliver separate results.  Besides, the checks are defined to be run on pull request, so I expect that such checks' results are bound to each pull request and not to the last commit in source branch.  Sadly my experience has shown that it is the latter.
